src/webapp/components/module-list-table/useModuleList.tsx

```diff
@@ -0,0 +1,649 @@
+                onSave: async value => {
+                    updateMarkdownDialog(null);
+                    if (!tableActions.editContents || !row.value || !row.moduleId) return;
+
+                    await tableActions.editContents({ id: row.moduleId, text: row.value, value });
+                    await refreshRows();
+                },
+            });
+        },
+        [tableActions, rows, refreshRows]
+    );
+
+    const installApp = useCallback(
+        async (ids: string[]) => {
+            if (!tableActions.installApp || !ids[0]) return;
+
+            loading.show(true, i18n.t("Installing application"));
+            const installed = await tableActions.installApp({ id: ids[0] });
+            loading.reset();
+
+            if (!installed) {
+                snackbar.error("Error installing app");
+                return;
+            }
+
+            snackbar.success("Successfully installed app");
+            await refreshRows();
+        },
+        [tableActions, snackbar, loading, refreshRows]
+    );
+
+    const resetModules = useCallback(
+        (ids: string[]) => {
+            updateDialog({
+                title: i18n.t("Are you sure you want to reset selected modules to its default value?"),
+                description: i18n.t("This action cannot be reversed."),
+                onCancel: () => updateDialog(null),
+                onSave: async () => {
+                    updateDialog(null);
+                    if (!tableActions.resetModules) return;
+
+                    loading.show(true, i18n.t("Resetting modules to default value"));
+                    await tableActions.resetModules({ ids });
+                    loading.reset();
+
+                    snackbar.success(i18n.t("Successfully resetted modules to default value"));
+                    await refreshRows();
+                },
+                cancelText: i18n.t("Cancel"),
+                saveText: i18n.t("Reset app to factory settings"),
+            });
+        },
+        [tableActions, loading, refreshRows, snackbar]
+    );
+
+    const exportModule = useCallback(
+        async (ids: string[]) => {
+            if (!ids[0]) return;
+            loading.show(true, i18n.t("Exporting module(s)"));
+            await usecases.modules.export(ids);
+            loading.reset();
+        },
+        [loading, usecases]
+    );
+
+    const exportTranslations = useCallback(
+        async (ids: string[]) => {
+            if (!ids[0]) return;
+            loading.show(true, i18n.t("Exporting translations"));
+            await exportTranslation(() => usecases.modules.extractTranslations(ids[0]), ids[0]);
+            loading.reset();
+        },
+        [loading, usecases, exportTranslation]
+    );
+
+    const onTableChange = useCallback(({ selection }: TableState<ListItem>) => {
+        setSelection(selection);
+    }, []);
+
+    const actions: TableAction<ListItem>[] = useMemo(
```

Comment:
No need to refactor, but as a general note: when we have a custom hook with a lot of pieces, the typical abstraction guidelines hold, we should strive for small, focused functions. This can be done creating more custom hooks.